scripts/fast-reboot

```diff
@@ -659,6 +633,15 @@ if [[ "$REBOOT_TYPE" = "fast-reboot" ]]; then
         unload_kernel
         exit "${EXIT_COUNTERPOLL_DELAY_FAILURE}"
     fi
+
+    # Clear all routes except of default routes for faster reconciliation time.
+    sonic-db-cli APPL_DB eval "
+        for _, k in ipairs(redis.call('keys', '*')) do
+            if string.match(k, 'ROUTE_TABLE:') and not string.match(k, 'ROUTE_TABLE:0.0.0.0/0') and not string.match(k, 'ROUTE_TABLE:::/0') then \
```

Comment:
As Dror wrote it is similar to the previous implementation. Currently the ARP and FDB tables are dumped to files as default routes (by the filter_fdb_entries and the fast-reboot-dump scripts), now instead of dumping and reloading we will delete all non-default routes. In the new implementation we don't need to reload the routes after reset as the default routes are still present and it will shorten the dataplane downtime.

Reply:
I might be missing something, can you please point me to the present logic for `save only the default routes` in fast-reboot case?

Reply:
Saving of default routes was done by the fast-reboot-dump script other than that it was saving also ARP and FDB tables. It was being saved to /host/fast-reboot/default_routes.json

Reply:
The new suggested approach was tested side-by-side with the previous approach of dumping the arp/fdb/default_routes to json files and reloading them. This single change improves dataplane downtime by more than a second by itself.

Reply:
Sorry for the late reply here. I still don't understand why ARP and FDB tables are not dumped anymore. AFAIK, that is not the part of design doc for this change. 

Along with that it seems that:
`generate_media_config` call is also skipped. We shouldn't do that, as it was added to prevent some platform-specific issue like https://github.com/sonic-net/sonic-buildimage/issues/9165

`send_garp_nd` call is being skipped. I think this is also still needed as this will make sure ARP entries on downstream peers (Servers) are kept fresh and latest before proceeding to shut down syncd.

Reply:
Reverted this change, will use the current implementation of using fast-reboot-dump and filter_fdb_entries scripts.

Reply:
I inspected the last change made - reintroducing the use of fast-reboot-dump and filder_fdb_entries and the implications it had on fast-reboot performance and I wanted to rest you assured that it is not needed.



The change of having fast-reboot over warm-reboot infrastructure introduces the call to backup_database which wasn't used previously for fast-reboot. This function backups FDB_TABLE and APP_DB is being dumped as well to dump.rdb so also media settings are being stored and alongside with ARPs. This is being restored in the change made in the sonic-buildimage PR which should be merged together. In this case the issue [https://github.com/sonic-net/sonic-buildimage/issues/9165 won't](https://github.com/sonic-net/sonic-buildimage/issues/9165%C2%A0won%27t) reproduce.



The main change that the relevant data will be restored in a similar way to as it is being restored in warm-reboot scenario and not using fdb.json; arp.json; default_routes.json; media_config.json files.



If we will call backup_database function without deleting all routes except default routes will result in storing and then restoring all the routes after reset. This will affect dataplane downtime as the amount of routes increases and this is not a desired behavior.



The specific implementation wasn't explicitly described in the HLD but the use of warm-reboot infrastructure i.e. use backup_database and dump.rdb is the cause for this change as the current logic of storing arp/fdb/default_routes and media settings is not changed.

Reply:
I didn't read in detail. but the explanation sound reasonable to me.

Reply:
@tjchadaga can you please check this PR to ensure https://github.com/sonic-net/sonic-buildimage/issues/9165 will not be seen again after this new design change gets merged?

Reply:
@vaibhavhd - based on the design of backup and restore of app db using dump.rdb, i think media settings should be restored on time to avoid https://github.com/sonic-net/sonic-buildimage/issues/9165